### ISSUE TITLE
`ProcessOutput.exitCode` is `null` when process is terminated by signal

### ISSUE THREAD

**Issue by hankei6km** [google/zx#310]
## Background

I want to detect the signal that terminated the process executed from `$`.

## Expected Behavior

The type of `ProcessOutput.exitCode` is `number`(it is not contained `| null`).

Therefore, expects `ProcessOutput.exitCode` to return `143` when `ProcessPromise.kill ('SIGTERM')` is called.
(Alternatively, detects the signal that terminated the process. ie.`ProcessOutput.signal`)

https://github.com/google/zx/blob/6e5bebbd3c826c98bd4d92fb62f777055ec2ae8e/index.d.ts#L47-L53

## Actual Behavior

`ProcessOutput.exitCode` return `null`.

## Steps to Reproduce the Problem

```js
#!/usr/bin/env zx

const p = $`sleep 100`

p.then((v) => {
  console.log(`then: exitCode=${v.exitCode}`)
}).catch((r) => {
  console.log(`catch: exitCode=${r.exitCode}`)
})

setTimeout(() => {
  p.kill('SIGTERM')
}, 2 * 1000)
```

```console
$ zx exit-code-by-kill.mjs
catch: exitCode=null
```

## Specifications

- Version: zx version 5.1.0
- Platform: node v16.13.2(16-bullseye)

## Workaround

`$` has not received a signal.

https://github.com/google/zx/blob/6e5bebbd3c826c98bd4d92fb62f777055ec2ae8e/index.mjs#L90-L99

Therefore, handle the `close`(or `exit`) event of `ProcessPromise.child`.
https://nodejs.org/dist/latest-v16.x/docs/api/child_process.html#event-close
https://nodejs.org/dist/latest-v16.x/docs/api/child_process.html#event-exit
https://nodejs.org/dist/latest-v16.x/docs/api/child_process.html#subprocesskillsignal

```js
#!/usr/bin/env zx

$.verbose = false
const p = $`sleep 100`

p.then((v) => {
  console.log(`then: exitCode=${v.exitCode}`)
}).catch((r) => {
  console.log(`catch: exitCode=${r.exitCode}`)
})

p.child.on('close', (code, signal) => {
  console.log(`close: code=${code} signal=${signal}`)
})

setTimeout(() => {
  p.kill('SIGTERM')
}, 2 * 1000)
```

```console
$ zx exit-code-by-kill.mjs 
close: code=null signal=SIGTERM
catch: exitCode=null
```

**Comment by antonmedv** [comment]
Fixed in 53a215e99e0c7f732cee6b09d1bb55eb53e3cc89
Now there is singal field. 
```
$ while true; do :; done
Error:
    at file:///test.mjs:2:12
    exit code: null
    signal: SIGKILL
```

**Comment by hankei6km** [comment]
Thank you for the update.
It is very useful.
